includes/class-reader-activation.php

```diff
@@ -0,0 +1,323 @@
+
+	/**
+	 * Reader user meta keys.
+	 */
+	const READER         = 'np_reader';
+	const EMAIL_VERIFIED = 'np_reader_email_verified';
+
+	/**
+	 * Initialize hooks.
+	 */
+	public static function init() {
+		if ( self::is_enabled() ) {
+			\add_action( 'clear_auth_cookie', [ __CLASS__, 'clear_auth_intention_cookie' ] );
+			\add_action( 'set_auth_cookie', [ __CLASS__, 'clear_auth_intention_cookie' ] );
+			\add_filter( 'login_form_defaults', [ __CLASS__, 'add_auth_intention_to_login_form' ], 20 );
+			\add_action( 'resetpass_form', [ __CLASS__, 'set_reader_verified' ] );
+			\add_action( 'password_reset', [ __CLASS__, 'set_reader_verified' ] );
+		}
+	}
+
+	/**
+	 * Whether reader activation is enabled.
+	 *
+	 * @return bool True if reader activation is enabled.
+	 */
+	public static function is_enabled() {
+		$is_enabled = defined( 'NEWSPACK_EXPERIMENTAL_READER_ACTIVATION' ) && NEWSPACK_EXPERIMENTAL_READER_ACTIVATION;
+
+		/**
+		 * Filters whether reader activation is enabled.
+		 *
+		 * @param bool $is_enabled Whether reader activation is enabled.
+		 */
+		return \apply_filters( 'newspack_reader_activation_enabled', $is_enabled );
+	}
+
+	/**
+	 * Add auth intention email to login form defaults.
+	 *
+	 * @param array $defaults Login form defaults.
+	 *
+	 * @return array
+	 */
+	public static function add_auth_intention_to_login_form( $defaults ) {
+		$email = self::get_auth_intention();
+		if ( ! empty( $email ) ) {
+			$defaults['label_username'] = __( 'Email address', 'newspack' );
+			$defaults['value_username'] = $email;
+		}
+		return $defaults;
+	}
+
+	/**
+	 * Clear the auth intention cookie.
+	 */
+	public static function clear_auth_intention_cookie() {
+		/** This filter is documented in wp-includes/pluggable.php */
+		if ( ! apply_filters( 'send_auth_cookies', true ) ) {
+			return;
+		}
+
+		// phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.cookies_setcookie
+		setcookie( self::AUTH_INTENTION_COOKIE, ' ', time() - YEAR_IN_SECONDS, COOKIEPATH, COOKIE_DOMAIN );
+	}
+
+	/**
+	 * Set the auth intention cookie.
+	 *
+	 * @param string $email Email address.
+	 */
+	public static function set_auth_intention_cookie( $email ) {
+		/** This filter is documented in wp-includes/pluggable.php */
+		if ( ! apply_filters( 'send_auth_cookies', true ) ) {
+			return;
+		}
+
+		/**
+		 * Filters the duration of the auth intention cookie expiration period.
+		 *
+		 * @param int    $length Duration of the expiration period in seconds.
+		 * @param string $email  Email address.
+		 */
+		$expire = time() + \apply_filters( 'newspack_auth_intention_expiration', 30 * DAY_IN_SECONDS, $email );
+		// phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.cookies_setcookie
+		setcookie( self::AUTH_INTENTION_COOKIE, $email, $expire, COOKIEPATH, COOKIE_DOMAIN, true );
+	}
+
+	/**
+	 * Get the auth intention.
+	 *
+	 * @return string|null Email address or null if not set.
+	 */
+	public static function get_auth_intention() {
```

Comment:
`set_auth_intention` was renamed to `set_auth_intention_cookie` in 7242930c6d775d3f9502c63ccfffe3f8c2220d2f – the getter method should also be renamed.

Reply:
I'd rather not in this case. This is currently the primary and only getter of the auth intention. The fact that the getter currently uses a cookie is irrelevant to the consumer of this exposed method.

If we decide to change the storage to something else (not a cookie), it would require this one method to be renamed across different codebases, as we intend to use the auth intention for several UX improvements.

Reply:
From a developer perspective I immediately wonder what is "auth intention". The concept itself is not a standard one (like e.g. login), and using its name in code as if it was a code entity adds to the confusion – is it a class, a DB field, does this method make a request to fetch the intention?. 
`get_auth_intention_cookie` is on the other hand understandable – get the cookie value that handles a feature called "auth intention". 
I understand the value of the name being generic and not tied to an implementation detail. How about giving it an understandable, generic name, though, like `get_auth_intention_identifier`/`get_auth_intention_value`?

Reply:
Thank you for your thoughts on this! I agree that we can use `get_auth_intention_value()` in order to give more clarity and still preserve its agnostic aspect towards storage.

09dc78b904972e0ee16b42f842a0c325b806da77 updates the name